channel.go

```diff
@@ -60,17 +63,71 @@ func (c *channel) IsSubscribed() bool {
 	return c.subscribed
 }
 
-func (c *channel) Subscribe() error {
-	c.mutex.RLock()
-	defer c.mutex.RUnlock()
+type subscribeOptions struct {
+	successTimeout time.Duration
+}
 
-	if c.subscribed {
+// SubscribeOption is a configuration option for subscribing to a channel
+type SubscribeOption func(*subscribeOptions)
+
+const defaultSuccessTimeout = 10 * time.Second
+
+// WithSuccessTimeout returns a SubscribeOption that sets the time that a subscription
+// request will wait for a success response from Pusher before timing out. The
+// default is 10 seconds.
+func WithSuccessTimeout(d time.Duration) SubscribeOption {
+	return func(o *subscribeOptions) {
+		o.successTimeout = d
+	}
+}
+
+// ErrTimedOut is the error returned when there is a timeour waiting for a subscription
+// confirmation from Pusher
+var ErrTimedOut = errors.New("timed out")
+
+func (c *channel) sendSubscriptionRequest(data subscribeData, o *subscribeOptions) error {
+	c.mutex.Lock()
+	c.subscribeSuccess = make(chan struct{})
+	c.mutex.Unlock()
+
+	doneChan := make(chan error)
+
+	go func() {
+		var err error
+		select {
+		case <-c.subscribeSuccess:
+			err = nil
+		case <-time.After(o.successTimeout):
+			err = ErrTimedOut
+		}
+		select {
+		case doneChan <- err:
+		default:
```

Comment:
Please add a comment concerning the intended pass-through default behavior.  It's slightly unclear as it stands.